src/main/credentialServer/httpCredentialServer.ts

```diff
@@ -0,0 +1,210 @@
+      }
+      const command = this.lookupCommand(method, pathParts[0]);
+
+      if (!command) {
+        console.log(`404: No handler for URL ${ method } ${ path }.`);
+        response.writeHead(404, { 'Content-Type': 'text/plain' });
+        response.write(`Unknown command: ${ method } ${ path }`);
+
+        return;
+      }
+      await command.call(this, helperName, data, request, response);
+    } catch (err) {
+      console.log(`Error handling ${ request.url }`, err);
+      response.writeHead(500, { 'Content-Type': 'text/plain' });
+      response.write('Error processing request.');
+    } finally {
+      response.end();
+    }
+  }
+
+  protected lookupCommand(method: string, commandName: string): dispatchFunctionType|undefined {
+    if (commandName) {
+      return this.dispatchTable[method]?.[commandName];
+    }
+
+    return undefined;
+  }
+
+  async get(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireJSONOutput, helperName, 'get', data, request, response);
+  }
+
+  async list(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireJSONOutput, helperName, 'list', data, request, response);
+  }
+
+  async erase(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'erase', data, request, response);
+  }
+
+  async store(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'store', data, request, response);
+  }
+
+  protected combineStrings(stdout: string, stderr: string): string {
+    if (!stdout) {
+      return stderr;
+    } else if (!stderr) {
+      return stdout;
+    } else {
+      return `${ stdout }\n\n${ stderr }}`;
+    }
+  }
+
+  protected async doNamedCommand(outputChecker: checkFunctionOutputType, helperName: string, commandName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    let stdout: string;
+    let stderr: string;
+
+    try {
+      ( { stdout, stderr } = this.runSyncInput(data, helperName, [commandName]));
+      if (outputChecker(stdout, stderr)) {
+        response.writeHead(200, { 'Content-Type': 'text/plain' });
+        response.write(stdout);
+
+        return await Promise.resolve();
+      }
+      stderr = this.combineStrings(stdout, stderr);
+    } catch (err: any) {
+      stderr = this.combineStrings(err.stdout ?? '', err.stderr ?? '');
+    }
+    console.debug(`credentialServer: ${ commandName }: writing back status 400, error: ${ stderr }`);
+    response.writeHead(400, { 'Content-Type': 'text/plain' });
+    response.write(stderr);
+
+    return await Promise.resolve();
```

Comment:
This method doesn't actually do anything async (but it _should_); ideally, make the child process async, but alternatively drop the `async`?

Reply:
Now using an async child process.